### PR TITLE
[PROD-795] Let swap page display the price info when pool is unavailable

### PR DESCRIPTION
**Pull request by leqiang-deltafi**
Using our current dropdown + swith token combination, it is possible to select a pair of tokens which does not have a pool. In this case, we should show the swap button as unavailable but should still able to show the price information

Current:
Screen Shot 1401-04-03 at 13 53 19

New:
Screen Shot 1401-04-03 at 13 53 22

### CODE REVIEW

File: src/constants/deployConfigV2.ts [hunk 1]
```diff
@@ -57,6 +57,30 @@ export function getPoolConfigBySymbols(baseSymbol: String, quoteSymbol: String):
   );
 }
 
+// if the pool doesn't exist in our deployment
+// we will return a fake poolConfig with undefined address info but with symbols
+// base and quote token in this fake poolConfig are in alphabetical orde
+export function getPoolConfigBySymbolsAllowFake(
```
Comment (leqiang-deltafi) [inline src/constants/deployConfigV2.ts:63]: allow returning fake config without address if real config doesn't exist

Comment (ypeng-deltafi) [reply src/constants/deployConfigV2.ts:63]: I would prefer we don't include this mock config at this level. can you just handle it in the `pool`

File: src/views/Swap/Home.tsx [hunk 2]
```diff
@@ -268,7 +268,7 @@ const Home: React.FC = (props) => {
   const tokenFrom = swapView.tokenFrom;
   const tokenTo = swapView.tokenTo;
 
-  const poolConfig: PoolConfig = getPoolConfigBySymbols(
+  const poolConfig: PoolConfig = getPoolConfigBySymbolsAllowFake(
```
Comment (ypeng-deltafi) [inline src/views/Swap/Home.tsx:271]: can we just use `getPoolConfigBySymbols() || mockPoolConfig(base, quote)`?

Comment (ypeng-deltafi) [reply src/views/Swap/Home.tsx:271]: I am not comfortable to add a hack just for this page as a general utils

Comment (leqiang-deltafi) [reply src/views/Swap/Home.tsx]: updated with this solution

File: src/views/Swap/Home.tsx [hunk 3]
```diff
@@ -292,15 +292,15 @@ const Home: React.FC = (props) => {
 
   const { marketPrice, basePrice, quotePrice } = useSelector(selectMarketPriceByPool(poolConfig));
   const exchangeRateLabel = useMemo(() => {
-    if (basePrice && quotePrice && swapInfo) {
+    if (basePrice && quotePrice) {
```
Comment (ypeng-deltafi) [inline src/views/Swap/Home.tsx:295]: if there is no pool, I feel we should not show the exchange rate.

Comment (ypeng-deltafi) [reply src/views/Swap/Home.tsx:295]: can we just show the pyth price?

Comment (leqiang-deltafi) [reply src/views/Swap/Home.tsx:295]: ok

Comment (leqiang-deltafi) [reply src/views/Swap/Home.tsx]: updated

File: src/constants/deployConfigV2.ts [hunk 4]
```diff
@@ -57,6 +57,19 @@ export function getPoolConfigBySymbols(baseSymbol: String, quoteSymbol: String):
   );
 }
 
+// mock config for price display
+export function getMockedPoolConfigBySymbol(baseSymbol: String, quoteSymbol: String): PoolConfig {
```
Comment (ypeng-deltafi) [inline src/constants/deployConfigV2.ts:61]: I think you can keep it in the pool page. no need to be here

**Review by ypeng-deltafi — APPROVED** [review]
lgtm with nits

File: src/constants/deployConfigV2.ts (hunk 4 above)
Comment (leqiang-deltafi) [reply src/constants/deployConfigV2.ts]: moved to the pool page. we can move to the util if somewhere else also need it

**Pull request merged**